File: DisturbedPeople/awnser.py
from __future__ import annotations

from dataclasses import dataclass
from typing import List


@dataclass
class DisturbedPeopleSolver:
    flats: List[int]

    def minimum_switch_offs(self) -> int:
        state = self.flats[:]
        n = len(state)
        operations = 0

        for i in range(1, n - 1):
            if state[i - 1] == 1 and state[i] == 0 and state[i + 1] == 1:
                state[i + 1] = 0
                operations += 1

        return operations


def main() -> None:
    import sys

    data = sys.stdin.read().split()
    if not data:
        return

    n = int(data[0])
    flats = list(map(int, data[1:1 + n]))

    solver = DisturbedPeopleSolver(flats=flats)
    result = solver.minimum_switch_offs()
    print(result)


if __name__ == "__main__":
    main()

File: DisturbedPeople/awnser.go
package main

import (
    "bufio"
    "fmt"
    "os"
)

type DisturbedPeopleSolver struct {
    Flats []int
}

func (s DisturbedPeopleSolver) MinimumSwitchOffs() int {
    flats := make([]int, len(s.Flats))
    copy(flats, s.Flats)

    n := len(flats)
    operations := 0

    for i := 1; i+1 < n; i++ {
        if flats[i-1] == 1 && flats[i] == 0 && flats[i+1] == 1 {
            flats[i+1] = 0
            operations++
        }
    }

    return operations
}

func main() {
    in := bufio.NewReader(os.Stdin)
    out := bufio.NewWriter(os.Stdout)
    defer out.Flush()

    var n int
    if _, err := fmt.Fscan(in, &n); err != nil {
        return
    }

    flats := make([]int, n)
    for i := 0; i < n; i++ {
        fmt.Fscan(in, &flats[i])
    }

    solver := DisturbedPeopleSolver{Flats: flats}
    result := solver.MinimumSwitchOffs()
    fmt.Fprintln(out, result)
}
